qiskit/backends/ibmq/ibmqbackend.py

```diff
@@ -101,9 +98,9 @@ def run(self, q_job):
             except (KeyError, TypeError):
                 hpc = None
 
-        # TODO: this should be self._configuration['name'] - need to check that
-        # it is always the case.
         backend_name = qobj['config']['backend_name']
+        assert backend_name == self.name, \
```

Comment:
Can you `raise` instead of `assert`, for consistency with the rest of the codebase?